builtin/logical/pki/path_tidy.go

```diff
@@ -872,7 +903,11 @@ func (b *backend) doTidyCertStore(ctx context.Context, req *logical.Request, log
 
 		cert, err := x509.ParseCertificate(certEntry.Value)
 		if err != nil {
-			return fmt.Errorf("unable to parse stored certificate with serial %q: %w", serial, err)
+			// if tidying invalid certs is not enabled, throw warning
+			if !config.InvalidCerts {
+				logger.Warn("unable to parse stored certificate with serial %q: %w; tidy by enabling tidy_invalid_certs", serial, err)
+			}
```

Comment:
In particular, I'd make this logic something like:

```suggestion
			if !config.InvalidCerts && !haveWarned {
				logger.Warn("unable to parse stored certificate with serial %q: %w; tidy by enabling tidy_invalid_certs", serial, err)
				haveWarned = true
			} else {
				if err := req.Storage.Delete(ctx, "certs/"+serial); err != nil {
					return fmt.Errorf("error deleting invalid certificate %s: %w", serial, err)
				}
				b.tidyStatusIncCertStoreCount()
			}
```

or something. :D